### ISSUE TITLE
Illegal back reference when creating new link (pressing 'Enter')

### ISSUE THREAD

**Issue by jeromg** [vimwiki/vimwiki#970]
* Environment:
  - Vimwiki Stable version: 2.5
  - NVIM v0.5.0-619-g1153ac903
  - Windows 10

* Vimwiki config
```
" VimWiki {{{
let g:vimwiki_list = [ { 'path': '~/wiki',
                     \    'path_html': '~/wiki/html',
                     \    'diary_sort': 'asc',
                     \    'diary_rel_path': 'journal',
                     \    'syntax': 'markdown',
                     \    'ext': '.md',
                     \ 'template_path': '~/wiki/templates',
                     \ 'template_default': 'default',
                     \ 'template_ext': '.htm',
                     \    'auto_tags': 1}
                     \ ]
let g:vimwiki_use_calendar=1
let g:vimwiki_folding='expr'
let g:vimwiki_table_mappings = 0
" 
" }}}
```
* Steps to reproduce
  - on a diary file (i.e. named "yyyy-mm-dd.md"), press \<Enter\> on any word
  - error message is:
```Error detected while processing function 
vimwiki#base#follow_link[89]..vimwiki#base#normalize_link[3]..vimwiki#markdown_base#normalize_link[5]..<SNR>94_normalize_link_syntax_n[48]..vimwiki#base#is_diary_file:
line    4:
E65: Illegal back reference
Press ENTER or type command to continue
```
  - Same problem on a mardown file named "2020-08-02T103016 - some note.md" (which is a file format I use once in a while)
  - Note: No problem occurs on a file that **does not start with an ISO8601 date**

Many thank in advance for your help (and by the way, I love vimwiki, thanks for the great job)

**Comment by tinmarino** [comment]
I would like the output of `:echom vimwiki#diary#get_diary_files()` I think there are links with spaces.
From your tracelog, the fix seems to be from :

https://github.com/vimwiki/vimwiki/blob/619f04f89861c58e5a6415a4f83847752928252d/autoload/vimwiki/base.vim#L2166

Which should be replaced by:
```
 \ filter(l:normalised_file_paths, 'v:val =~# "a:filename"')
```
or even better
```
 \ filter(l:normalised_file_paths, 'v:val =~# ''a:filename''')
```
(with 2 `'` instead of one `"`)

If that change work for you, please create a pull request, no test needed for this one so only 2-4 bytes added.
Cheers.

**Comment by tinmarino** [comment]
Ah and you did not do anything wrong, you just found a bug in the code (our/my bad: trying to use some functional programming ...)

**Comment by jeromg** [comment]
Thank you @tinmarino 
I've created PR #972 (my first ever PR on github!)
Thank you to the community here for vimwiki. I rely daily on it.

**Comment by tinmarino** [comment]
> I've created PR #972 (my first ever PR on github!)

The travel of a thousands miles begins with one step (Lao Tzi). Welcome! See comments there, it does not pass the test, I just fetched your PR but did not investiate try possible solutions.

**Comment by tinmarino** [comment]
Fixed: #974 Good job @jeromg
